sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/DataSourceV2Reader.java

```diff
@@ -46,6 +46,9 @@
  * Spark first applies all operator push-down optimizations that this data source supports. Then
  * Spark collects information this data source reported for further optimizations. Finally Spark
  * issues the scan request and does the actual data reading.
+ *
+ * This reader must be able to get constructed and serve readSchema() without assuming an active
+ * Spark session. An active session can be assumed when creating read tasks.
```

Comment:
Why this assumption/restriction?

Reply:
same question

Reply:
A reader has to be instantiated in DataStreamReader so we can get the schema.

When I tried doing this in implementation, my unit test failed reporting there was no current Spark session. I had assumed this was a real problem, but maybe it was just a test artifact? (Or maybe there's a way to set the DataStreamReader's session as "current" for just the instantiation of the reader?)

Reply:
Discussed offline and it was indeed a test artifact. Removing this comment.